File: t001178.py
import math


def ek(f1, f2):
    d = abs(f1 - f2)
    if d <= 0.00000001:
        return True
    else:
        return False


t = int(input())
for i in range(t):
    s = [float(j) for j in input().split()]
    if ek(s[4], 0):
        print("0.00")
    else:
        det = 4 * (s[0] - s[2]) ** 2 + 8 * (s[1] - s[3]) * s[4]
        a = s[1] - s[3]
        b = 2 * (s[0] - s[2])
        if det < 0:
            print("Drong is strong.")
        else:
            if ek(a, 0):
                if ek(b, 0):
                    print("Drong is strong.")
                else:
                    x = (2 * s[4]) / b
                    print("%.2f" % x)
            else:
                x1 = (-b + math.sqrt(det)) / (2 * a)
                x2 = (-b - math.sqrt(det)) / (2 * a)
                x = max(x1, x2)
                if x >= 0:
                    print("%.2f" % x)
                else:
                    print("Drong is strong.")

File: t001178.cpp

#include <stdio.h>
#include <math.h>
int main()
{
	int n;
	float v0,v1,a0,a1,x,t,a,b,dert,x1,x0;
	scanf("%d",&n);
	while(n--)
	{
		scanf("%f %f %f %f %f",&v0,&a0,&v1,&a1,&x);
		if(x==0)
		{
			printf("0.00\n");
			continue;
		}
		if(v0<=v1&&a0<=a1)
		{
			printf("Drong is strong.\n");
			continue;
		}
		if(v0<v1&&a0>a1||v0>v1&&a0>=a1||v0==v1&&a0>a1)
		{
			a=0.5*a0-0.5*a1,b=v0-v1;
			dert=sqrt(b*b+4*a*x);
			t=(-b+dert)/(2*a);
			printf("%.2f\n",t);
			continue;
		}
		if(v0>v1&&a0<a1)
		{
			t=(v0-v1)/(a1-a0);
			x0=v0*t+0.5*a0*t*t;
			x1=v1*t+0.5*a1*t*t;
			if(x0==x1+x)
				printf("%.2f\n",t);
			else if(x0>x1+x)
			{
				a=0.5*a0-0.5*a1,b=v0-v1;
			    dert=sqrt(b*b+4*a*x);
			    t=(-b+dert)/(2*a);
			    printf("%.2f\n",t);
			}
			else
				printf("Drong is strong.\n");
		}
	}
	return 0;
}      